servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -60,18 +62,17 @@ def check(self, instance):
         try:
             self.start_snapshot()
             relation_types = self._process_and_cache_relation_types(instance_config, timeout)
-            self._process_components(instance_config, timeout)
+            self._process_components(instance_config, batch_size, timeout)
             self._process_component_relations(instance_config, batch_size, timeout, relation_types)
+            # Report ServiceCheck OK
+            msg = "ServiceNow CMDB instance detected at %s " % base_url
+            tags = ["url:%s" % base_url]
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=tags, message=msg)
```

Comment:
Call `self.stop_snapshot()` first. Only if that succeeds do we know for sure that the service check has ran its course correctly.

Reply:
stop_snapshot moved before OK service check call